snapshots/README.md

```diff
@@ -0,0 +1,236 @@
+# State Sync Snapshotting
+
+The `snapshots` package implements automatic support for Tendermint state sync
+in Cosmos SDK-based applications. State sync allows a new node joining a network
+to simply fetch a recent snapshot of the application state instead of fetching
+and applying all historical blocks. This can reduce the time needed to join the
+network by several orders of magnitude (e.g. weeks to minutes), but the node
+will not contain historical data from previous heights.
+
+This document describes the Cosmos SDK implementation of the ABCI state sync
+interface, for more information on Tendermint state sync in general see:
+
+* [Tendermint Core State Sync for Developers](https://medium.com/tendermint/tendermint-core-state-sync-for-developers-70a96ba3ee35)
+* [ABCI State Sync Spec](https://docs.tendermint.com/master/spec/abci/apps.html#state-sync)
+* [ABCI State Sync Method/Type Reference](https://docs.tendermint.com/master/spec/abci/abci.html#state-sync)
+
+## Overview
+
+For an overview of how Cosmos SDK state sync is set up and configured by
+developers and end-users, see the
+[Cosmos SDK State Sync Guide](https://blog.cosmos.network/cosmos-sdk-state-sync-guide-99e4cf43be2f).
+
+Briefly, the Cosmos SDK takes state snapshots at regular height intervals given
```

Comment:
Are these directories relative to the `datadir` provided to the app and Tendermint?

Reply:
They're relative to the "home" dir, as it's called in the code (not the user's homedir, which gets confusing) -- i.e. the parent of the `datadir`.

Reply:
`<home>/data` makes me think of `$HOME/data`. Maybe we can rename to `<node_home>/data`? I'm using this name:

https://github.com/cosmos/cosmos-sdk/blob/a15d7e31d42c77dd653d885f7fbd802f12cdf46c/simapp/app.go#L94-L95

Reply:
Yep, good idea -- done.